ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_submission_service.proto

```diff
@@ -37,11 +37,17 @@ service CommandSubmissionService {
   // Errors:
   // - ``UNAUTHENTICATED``: if the request does not include a valid access token
   // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
-  // - ``NOT_FOUND``: if the request does not include a valid ledger id
+  // - ``NOT_FOUND``: if the request does not include a valid ledger id or the submission rank or
+  //   deduplication period are too old.
   // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
   // - ``UNAVAILABLE``: if the participant is not yet ready to submit commands or if the service has been shut down.
   // - ``RESOURCE_EXHAUSTED``: if the participant or the ledger is overloaded. Clients should back off exponentially and retry.
-  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources
+  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources or
+  //   if there is another submission in flight for the same change ID.
+  // - ``FAILED_PRECONDITION`` if there is another submission with a higher submission rank.
+  // - ``ALREADY_EXISTS`` if there is a definite completion for the same deduplication key with at least the same submission rank
```

Comment:
This seems to mix violations from deduplication with violations from the rank guarantee. It would make sense to allow reporting deduplication violations synchronously. For rank violations, the async route would be better as it generates a new completion offset.

Reply:
Double-check condition `at least the same submission rank`